js/ui/codeView.js

```diff
@@ -106,6 +106,47 @@ function _getWindowId(win) {
     return ('window:%d').format(win.get_stable_sequence());
 };
 
+const _ensureHackDataFile = (function () {
+    let keyfile = new GLib.KeyFile();
+    let initialized = false;
+
+    const flatpakPath = '/var/lib/flatpak';
+    const fileRelPath = 'share/hack-components/hack-data.ini';
+    const filePath = `${flatpakPath}/app/com.endlessm.HackComponents/current/active/files/${fileRelPath}`;
+
+    return function() {
+        if (initialized)
+            return keyfile;
+
+        try {
+            keyfile.load_from_file(filePath, GLib.KeyFileFlags.NONE);
+        } catch (err) {
+            logError(err, 'Error reading hack data file');
```

Comment:
We should catch the case here where the file does not exist, and avoid printing an error in that case